clients/cobol-lsp-vscode-extension/src/services/snippetcompletion/cobolSnippets.json

```diff
@@ -2609,58 +2609,278 @@
   },
-  "SEARCH ALL WHEN": {
-    "prefix": "SEARCH ALL WHEN",
+  "SAME": {
+    "prefix": "SAME",
+    "body": ["SAME ${1|AREA,SORT,SORT-MERGE|} FOR ${2:filename}."],
+    "description": "ENVIRONMENT DIVISION / INPUT-OUTPUT SECTION / I-O-CONTROL - The SAME clause is syntax checked but not executable."
+  },
+  "SAME RECORD AREA": {
+    "prefix": "SAME RECORD AREA",
+    "body": ["SAME RECORD AREA FOR ${1:filename}."],
+    "description": "ENVIRONMENT DIVISION / INPUT-OUTPUT SECTION / I-O-CONTROL - The SAME RECORD AREA clause specifies that the same main storage be used for two or more files."
+  },
+  "SD": {
+    "prefix": "SD",
+    "body": [
+      "SD ${1:SORT-FILE}",
+      "\t\tLABEL RECORDS ARE STANDARD",
+      "\t\tBLOCK CONTAINS 0",
+      "\t\tDATA RECORD IS ${2:SORT-RECORD}.",
+      "01 ${3:SORT-RECORD} ${4:PIC X(1324)}."
+    ],
+    "description": "DATA DIVISION - The SD (SORT DESCRIPTION) defines files that are to be sorted and that are referred to in your PROCEDURE DIVISION's SORT statement."
+  },
+  "SEARCH": {
+    "prefix": "SEARCH",
+    "body": [
+      "SET SUB TO 1.",
+      "\tSEARCH ${1:WS-TABLE}",
+      "\tAT END DISPLAY \"${2:ELEMENT NOT FOUND IN TABLE}\"",
+      "\tWHEN ${1:WS-TABLE}(SUB) = ${3:WS-SRCH-ELEMENT}",
+      "\tDISPLAY \"FOUND IN TABLE\"",
+      "END-SEARCH."
+    ],
+    "description": "SEARCH searches a table SEQUENTIALLY (top to bottom) for an element that satisfies a condition. The SEARCH statement automatically increments the table subscript."
+  },
+  "SEARCH ALL": {
+    "prefix": "SEARCH ALL",
     "body": [
-      "SEARCH ALL ${1:item} WHEN ${2:item1} EQUAL ${3:item3} ${4:item4}"
+      "SEARCH ALL ${1:WS-TABLE}",
+      "\tAT END DISPLAY \"${2:ELEMENT NOT FOUND IN TABLE}\"",
+      "\tWHEN ${1:WS-TABLE}(IDX) = ${3:WS-SRCH-ELEMENT}",
+      "\tDISPLAY \"FOUND IN TABLE\"",
+      "END-SEARCH."
     ],
-    "description": "SEARCH ALL item1 WHEN item2 EQUAL item3 item4 "
+    "description": "SEARCH searches a table SEQUENTIALLY (top to bottom) for an element that satisfies a condition. The SEARCH statement automatically increments the table subscript."
   },
   "SEARCH WHEN": {
     "prefix": "SEARCH WHEN",
     "body": ["SEARCH ${1:item} WHEN ${2:condition} ${3:item}"],
     "description": "SEARCH WHEN condition item"
   },
+  "SEGMENT-LIMIT IS": {
+    "prefix": "SEGMENT-LIMIT IS",
+    "body": ["SEGMENT-LIMIT IS ${1:...}"],
+    "description": "A reserved word used in early IBM compilers. SEGMENT-LIMIT is syntax checked, but not operational."
+  },
```

Comment:
`SEGMENT-LIMIT is syntax checked, but not operational.` => we don't need the snippet then.
```suggestion
```